db/hints/manager.cc

```diff
@@ -269,6 +284,19 @@ sync_point::shard_rps manager::calculate_current_sync_point(std::span<const gms:
 }
 
 future<> manager::wait_for_sync_point(abort_source& as, const sync_point::shard_rps& rps) {
+    using lock_type = std::shared_lock<seastar::shared_mutex>;
+    // Prevent the migration to host-ID-based hinted handoff until this function finishes its execution.
+    const auto shared_lock = co_await std::invoke(coroutine::lambda([&] () -> future<lock_type> {
+        co_await _migration_mutex.lock_shared();
```

Comment:
Nit: I think it would be a bit nicer to use `seastar::with_shared` here instead.

Not worth fixing it now.